src/metadata/database.rs

```diff
@@ -258,4 +259,9 @@ impl Database {
             .get(code)
             .map(|m| m.iter().map(AsRef::as_ref).collect())
     }
+
+    /// Iterator over all `Metadata` entries in this database.
+    pub fn iter(&self) -> impl Iterator<Item = &Arc<Metadata>> {
```

Comment:
I'm wondering whether `&Arc<Metadata>` is the actual thing we'd want to expose here. If we ever decide to change the internal representation, it'd break a public interface.

Reply:
The thing is that this should be cheap. I.e. I don't want a copy of the DB. I want access to the DB.
I can also put this behind a feature flag that is off by default.

Reply:
> The thing is that this should be cheap. I.e. I don't want a copy of the DB.

It definitely should be. Should we maybe AsRef them? I'd think returning `impl Iterator<Item = &Metadata>` should be possible through a map, but I'm not sure.

Reply:
> It definitely should be. Should we maybe AsRef them? I'd think returning `impl Iterator<Item = &Metadata>` should be possible through a map, but I'm not sure.

Did that in my latest commit. Works as expected.

Reply:
Cool! Let's hear what @gferon still has to say then :)